### PR TITLE
Add `rateslib` to packages

### PR DESCRIPTION
**Pull request by NickAltmann**
### Description

Adding meta for the rateslib package, a fixed income pricing library, see https://rateslib.readthedocs.io/en/latest/.

While the first release, 1.1, was pure python, subsequent releases have a rust extension, hence the need for a package here.

### Checklists

- [x] Add / update tests

### CODE REVIEW

File: packages/rateslib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  top-level:
+    - rateslib
+source:
+  url: https://files.pythonhosted.org/packages/02/d7/7695bb1b6a240ecfae77804d3fc242e30582fd83973f76dda03b23d921a6/rateslib-1.5.0.tar.gz
+  sha256: 7c7af2707f20974a70403689f84905757a4ed084e4c92229388d2fc30b331b04
+requirements:
+  host:
+    - numpy
+    - pandas
+    - matplotlib
```
Comment (ryanking13) [inline packages/rateslib/meta.yaml:13]: You don't need to set these as a host dependency.

File: packages/rateslib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  executable:
+    - rustup
+about:
+  home: https://github.com/attack68/rateslib
+  PyPI: https://pypi.org/project/rateslib
+  summary: A fixed income library for trading interest rates
+  license: Creative Commons Attribution, Non-Commercial, No-Derivatives 4.0 International Licence
+extra:
+  recipe-maintainers:
+    - attack68
```
Comment (ryanking13) [inline packages/rateslib/meta.yaml:27]: Could you please confirm that @attack68 would want to maintain the recipe?

**Review by ryanking13 — COMMENTED** [review]
Thanks. Could you sync your PR with the main branch? We've just deployed a fix for CI timeout.

File: packages/rateslib/meta.yaml (hunk 2 above)
Comment (NickAltmann) [reply packages/rateslib/meta.yaml]: Ok, attack68 is the developer of the rateslib library itself, but for now I've removed him as maintainer for this recipe.  I'll provide him with instructions, and should he wish to perform an update later he can add himself at that point.

File: packages/rateslib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    - numpy
+    - pandas
+    - matplotlib
+  executable:
+    - rustup
+about:
+  home: https://github.com/attack68/rateslib
+  PyPI: https://pypi.org/project/rateslib
+  summary: A fixed income library for trading interest rates
+  license: Creative Commons Attribution, Non-Commercial, No-Derivatives 4.0 International Licence
```
Comment (agriyakhetarpal) [inline packages/rateslib/meta.yaml:20]: Could we standardise this better by using the [SPDX short-form identifier](https://spdx.org/licenses/CC-BY-NC-ND-4.0.html) for this license?
```suggestion
  license: CC-BY-NC-ND-4.0
```

**Review by agriyakhetarpal — COMMENTED** [review]
Thanks for the contribution, @NickAltmann!

**Comment by agriyakhetarpal** [comment]
The failure can be adjusted either by shortening the test or by increasing the timeout value. I guess the test is already minimal, so the latter is better.

**Comment by hoodmane** [comment]
The suggestion more specifically is to annotate the test with something like `@pytest.mark.driver_timeout(60)` which would raise the timeout to 60 seconds.

**Review by agriyakhetarpal — APPROVED** [review]
Looks like it's working now, thanks, @NickAltmann!

**Pull request merged**